### ISSUE TITLE
[Navi21] Consider usage of `GetMaxHardwareComputeUnits()` instead of `GetMaxComputeUnits()`.

### ISSUE THREAD

**Issue by atamazov** [ROCm/MIOpen#1440]
For Navi21, the latter function returns a number of WGPs. The number of actual hardware CUs is twice bigger. The usage of number of WGPs when number of CUs is actually required may lead to underutilization of GPU.

The number of actual CUs is returned by `GetMaxHardwareComputeUnits()`. 

I recommend double-checking of all places where `GetMaxComputeUnits()` is currently used for Navi21 (https://github.com/ROCmSoftwarePlatform/MIOpen/search?q=GetMaxComputeUnits&type=code) and considering replacing it with `GetMaxHardwareComputeUnits()`. Note that this change affects only gfx10 targets (see #677 to find out why). 

Some specific places to look at:

- [ ] (No owner) @junliume @ce1adon 
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/00d57548c25399186eb201eb2e0e6f7480915c48/src/ocl/rnn_util_ocl.cpp#L63 https://github.com/ROCmSoftwarePlatform/MIOpen/blob/00d57548c25399186eb201eb2e0e6f7480915c48/src/ocl/rnn_util_ocl.cpp#L191

- [ ] @jerryyin 
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/f09132964d628ffb649f4a6ee14dadfc933a7a69/src/solver/mlir_common.cpp#L145

- [ ] @carlushuang 
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/f02d9598fc02ad4fbf67d4f1cd88ed4e296f0961/src/hip/batched_transpose_sol.cpp#L302

- [ ] @DrizztDoUrden @averinevg 
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/0183095329e7c1803e65de64350c7a057562d068/src/include/miopen/utility/transposing_solver.hpp#L177

- [ ] (No owner) @junliume 
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/d2b5a17cbd1d160ea107a78b68e641aa25b6b32b/src/ocl/mloNorm.cpp#L81

**Comment by carlushuang** [comment]
@atamazov [MIOpen/src/hip/batched_transpose_sol.cpp](https://github.com/ROCmSoftwarePlatform/MIOpen/blob/f02d9598fc02ad4fbf67d4f1cd88ed4e296f0961/src/hip/batched_transpose_sol.cpp#L302) will be used in any architecture, not only navi. Indeed this kernel is important in our NCHW xdlops support in gfx90a, by utilize this as NCHW<->NHWC transpose and re-use the NHWC kernel

**Comment by atamazov** [comment]
@carlushuang 
> [MIOpen/src/hip/batched_transpose_sol.cpp](https://github.com/ROCmSoftwarePlatform/MIOpen/blob/f02d9598fc02ad4fbf67d4f1cd88ed4e296f0961/src/hip/batched_transpose_sol.cpp#L302) will be used in any architecture, not only navi.

Of course, I know that. Replacing `GetMaxComputeUnits()` with `GetMaxHardwareComputeUnits()` affects only gfx10 targets (see #677).
